socorro/lib/search_common.py

```diff
@@ -310,6 +321,42 @@ def fix_hang_type_parameter(parameters):
                 hang_type.value = new_values
                 hang_type.data_type = 'int'
 
+    @staticmethod
+    def fix_version_parameter(parameters):
+        """Correct the version parameter.
+
+        If version is something finishing with a 'b' and operator is
+        'has terms', replace the operator with 'starts with' to query all
+        beta versions.
+        """
+        if 'version' in parameters:
+            marked_for_deletion = []
+            for version in parameters['version']:
+                if version.operator:
+                    # We only care about the '' operator ("has terms").
+                    continue
+
+                new_values = []
+                for val in version.value:
+                    if val.endswith('b'):
+                        parameters['version'].append(SearchParam(
+                            name='version',
+                            value=[val],
+                            data_type='str',
+                            operator='$',
+                            operator_not=version.operator_not,
+                        ))
+                    else:
+                        new_values.append(val)
+
+                if new_values:
+                    version.value = new_values
+                else:
+                    marked_for_deletion.append(version)
+
+            for param in marked_for_deletion:
+                parameters['version'].remove(param)
+
```

Comment:
What you're trying to achieve with the `marked_for_deletion` is easier done this way:
```python
for version in parameters['version'][:]:
      if some_reason(version):
          parameters['version'].remove(version)
```
So basically, by looping over a copy you can afford to do the `.remove()` thing. 

I had to double-check my argument. See [this snippet](https://gist.github.com/peterbe/04186d8ebc7c773c41ad) compared to [this one](https://gist.github.com/peterbe/efa61b928e612532755e) which is the wrong way to do it.